src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -202,6 +202,37 @@ std::string ClientImpl::make_user_agent_string(ClientConfig& config)
     return out.str();                                     // Throws
 }
 
+std::string ClientImpl::ReconnectInfo::get_message()
+{
+    static std::map<ConnectionTerminationReason, std::string> termination_reasons = {
+        {ConnectionTerminationReason::connect_operation_failed, "Connect operation failed"},
+        {ConnectionTerminationReason::closed_voluntarily, "Closed voluntarily"},
+        {ConnectionTerminationReason::read_or_write_error, "Read or write error"},
+        {ConnectionTerminationReason::ssl_certificate_rejected, "SSL certificate rejected"},
+        {ConnectionTerminationReason::ssl_protocol_violation, "SSL protocol violation"},
+        {ConnectionTerminationReason::websocket_protocol_violation, "Websocket protocol violation"},
+        {ConnectionTerminationReason::http_response_says_fatal_error, "HTTP response fatal error"},
+        {ConnectionTerminationReason::http_response_says_nonfatal_error, "HTTP response nonfatal error"},
+        {ConnectionTerminationReason::bad_headers_in_http_response, "HTTP respnose bad headers"},
+        {ConnectionTerminationReason::sync_protocol_violation, "Sync protocol violation"},
+        {ConnectionTerminationReason::sync_connect_timeout, "Sync connect timeout"},
+        {ConnectionTerminationReason::server_said_try_again_later, "Server response: try again later"},
+        {ConnectionTerminationReason::server_said_do_not_reconnect, "Server response: do not reconnect"},
+        {ConnectionTerminationReason::pong_timeout, "Pong timeout"},
+        {ConnectionTerminationReason::server_301_redirect, "HTTP 301 redirect"},
+        {ConnectionTerminationReason::missing_protocol_feature, "Missing protocol feature"}};
+
+    if (!m_reason) {
+        return "No reason";
+    }
+    if (auto&& search = termination_reasons.find(*m_reason); search != termination_reasons.end()) {
+        return search->second;
+    }
+    else {
```

Comment:
just use return directly instead of else branch

Reply:
Updated function